torch/fx/node.py

```diff
@@ -764,7 +764,8 @@ def is_impure(self) -> bool:
                 # impure since it mutates inputs
                 return True
 
-            if getattr(self.target, "_nondeterministic_seeded", False):
+            # intentionally avoiding getattr overhead here: https://github.com/pytorch/pytorch/issues/144775
+            if "_nondeterministic_seeded" in self.__dict__:
```

Comment:
I have a hard time believing that getattr overhead matters that much, especially because there isn't a custom getattr. But it's easy enough to test out this PR